### PR TITLE
Improvements to Key Vault REST spec

### PR DESCRIPTION
**Pull request by msfcolombo**
### Summary
- Renamed JsonWebKeyECName to JsonWebKeyCurveName.
- Restored order of key creation parameters to reduce compatibility issues.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - x ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by msfcolombo** [comment]
Please await @schaabs approval before merge.

**Review by schaabs — APPROVED** [review]
LGTM, @dsgouda can you please take a look at this we want to update this before any clients are generated from the original merge https://github.com/Azure/azure-rest-api-specs/pull/1538. The original order of parameters causes backwards breaking changes in the generated clients.

**Comment by dsgouda** [comment]
@vishrutshah has been assigned to take a look, please wait for him to respond.

**Comment by dsgouda** [comment]
Taking over since I had reviewed the private repo pr earlier.

File: specification/keyvault/data-plane/Microsoft.KeyVault/2016-10-01/keyvault.json
Comment (dsgouda) [inline specification/keyvault/data-plane/Microsoft.KeyVault/2016-10-01/keyvault.json]: While you are at it consider making a common reference instead of duplicating here.

**Review by dsgouda — APPROVED** [review]
Please reduce duplication here.

**Review by dsgouda — CHANGES_REQUESTED** [review]
Meant to request changes

**Comment by msfcolombo** [comment]
Although ECC are purely incremental changes from REST PoV, we have clients that can't handle different values for "kty". As per internal discussions, I'm aborting this change and @schaabs will revert the previous merge.

**Pull request closed without merging**
